owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/handlers.go

```diff
@@ -13,14 +14,40 @@ func HealthCheck(c echo.Context) error {
 	return c.String(http.StatusOK, "WORKING\n")
 }
 
+func ErrorMessage(c echo.Context) error {
+	return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
+}
+
 // GetTicket returns the userID ticket.
 func GetTicket(c echo.Context) error {
+
+	cookie, err := ReadCookie(c)
+
+	if err != nil {
+		return ErrorMessage(c)
+	}
+
+	type UserClaims struct {
+		Name string `json:"name"`
+		jwt.StandardClaims
+	}
+
+	token, err := jwt.ParseWithClaims(cookie, &UserClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return []byte("secret"), nil
+	})
+
+	claims, _ := token.Claims.(*UserClaims)
+
+	if !token.Valid {
+		return ErrorMessage(c)
```

Comment:
As you are using `ErrorMessage` in this if condition, your message "Permission denied" may not fit well here. 

Maybe it would be a good idea to send a super generic message to the user "Error getting this ticket" if you do want to use this `ErrorMessage` function in multiple cases.

It is alway good to keep in mind that sending very specific messages to users can lead to [User Enumeration](https://www.owasp.org/index.php/Testing_for_User_Enumeration_and_Guessable_User_Account_(OWASP-AT-002)) in some cases.

Reply:
All of these checks are about user permission. If one of these conditions are not correct, the user won't get the ticket because check fails for permission.